src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,244 @@
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Returns true if the execution is successful, otherwise returns false</returns>
+        [ContractMethod(0_01000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.ByteArray, ContractParameterType.ByteArray }, ParameterNames = new[] { "consignorPubKey", "consigneePubKey" })]
+        private StackItem DelegateOracleValidator(ApplicationEngine engine, Array args)
+        {
+            ECPoint consignorPubKey = args[0].GetSpan().AsSerializable<ECPoint>();
+            ECPoint consigneePubKey = args[1].GetSpan().AsSerializable<ECPoint>();
+            UInt160 account = Contract.CreateSignatureRedeemScript(consignorPubKey).ToScriptHash();
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
+            StoreView snapshot = engine.Snapshot;
+            StorageKey key = CreateStorageKey(Prefix_Validator, consignorPubKey);
+            if (snapshot.Storages.TryGet(key) != null)
+            {
+                StorageItem value = snapshot.Storages.GetAndChange(key);
+                value.Value = consigneePubKey.ToArray();
+            }
+            else
+            {
+                snapshot.Storages.Add(key, new StorageItem
+                {
+                    Value = consigneePubKey.ToArray()
+                });
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Get current authorized Oracle validator.
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Authorized Oracle validator</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Array)]
+        private StackItem GetOracleValidators(ApplicationEngine engine, Array args)
+        {
+            return new Array(engine.ReferenceCounter, GetOracleValidators(engine.Snapshot).Select(p => (StackItem)p.ToArray()));
+        }
+
+        /// <summary>
+        /// Get current authorized Oracle validator
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Authorized Oracle validator</returns>
+        public ECPoint[] GetOracleValidators(StoreView snapshot)
+        {
+            ECPoint[] consensusPublicKey = PolicyContract.NEO.GetValidators(snapshot);
+            IEnumerable<(ECPoint ConsensusPublicKey, ECPoint OraclePublicKey)> hasDelegateOracleValidators = GetDelegateOracleValidators(snapshot).Where(p => consensusPublicKey.Contains(p.ConsensusPublicKey));
+            foreach (var item in hasDelegateOracleValidators)
+            {
+                var index = System.Array.IndexOf(consensusPublicKey, item.ConsensusPublicKey);
+                if (index >= 0) consensusPublicKey[index] = item.OraclePublicKey;
```

Comment:
consensusPublicKey  -> oraclePubKeys
We can swap `hasDelegateOracleValidators` and `consensusPublicKey` in this `foreach` for performance.